File: src/main/java/com/example/backend/dto/tarot/TarotRequest.java
package com.example.backend.dto.tarot;

import com.example.backend.dto.common.Translatable;
import com.example.backend.lang.Language;
import java.util.List;

public record TarotRequest(
    List<TarotCardDto> cards,
    String text,
    Language from,
    Language to) implements Translatable {
}

File: src/main/kotlin/com/example/backend/dto/tarot/TarotRequest.kt
package com.example.backend.dto.tarot

import com.example.backend.dto.common.Translatable
import com.example.backend.lang.Language

data class TarotRequest(
    var cards: List<TarotCardDto>? = null,
    var text: String? = null,
    var from: Language? = null,
    var to: Language? = null
) : Translatable {

    override fun from(): Language? {
        return from
    }

    override fun to(): Language? {
        return to
    }

    override fun text(): String? {
        return text
    }
}
